ani-cli

```diff
@@ -632,8 +656,19 @@ if [ -z "$select_first" ]; then
 	choice="$REPLY"
 	case $choice in
 		n)
-			ep_choice_start=$((episode + 1))
-			ep_choice_end=""
+			if curl -s "$base_url/$anime_id-episode-$episode-5" | grep 404 > /dev/null; then
+				ep_choice_start=$((episode + 1))
+				ep_choice_end=""
+			else
+				if [ "$half_ep" -eq 1 ]; then
+					half_ep=0
+					ep_choice_start=$((episode + 1))
+					ep_choice_end=""
+				else
+					half_ep=1
+					ep_choice_start=$episode
+				fi
+			fi
```

Comment:
This approach can be improved.
What you are doing:
- checking if half ep of `$episode` exists
- incrementing by one if doesn't exist
- if it doesn't exist
  - increment it by one if half ep and remove half ep
  - add half ep if not half ep and add half ep
 
The better approach would be:
- check if `half_ep` equals 1, if yes then just increase `$episode` by one and set `half_ep` to 0.
- the rest remains ignoring `half_ep -eq 1` if part

So in the later approach you don't check if half ep exists if we are already on half ep, we will be saving an extra request.

Reply:
i think i fixed it now